torch/csrc/jit/script/convert_to_ssa.cpp

```diff
@@ -172,6 +172,62 @@ struct ControlFlowLoadStores {
   std::shared_ptr<TypeEnvironment> environment_stack = nullptr;
 };
 
+// The loop node is initially emitted as:
+// Loop(max_trip_count)
+//    block0(loop_counter) {
+//      <body>
+//    }
+//    block1 {
+//      <loop condition>
+//      -> (condition)
+//    }
+// Here, we inline the loop condition and convert the loop to the form:
+// Loop(max_trip_count, start_condition)
+//    block0(loop_counter, loop_carried_block*) {
+//      <body>
+//      -> (continue_condition)
+//    }
+
+void inlineLoopCondition(Node* n) {
+  Block* body_block = n->blocks().at(0);
+
+  auto pre_header = n->blocks().at(1);
+  auto header_block = n->addBlock();
+  header_block->cloneFrom(pre_header, [](Value* v) { return v; });
```

Comment:
Can we make `insertBlockCopy` into a public function (currently in loop unroller) and use it here? Basically does the same thing but is a self-contained function.